public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts

```diff
@@ -11,7 +11,23 @@ import { getDashboardSceneFor, getPanelIdForVizPanel, getQueryRunnerFor } from '
 import { DashboardScene } from './DashboardScene';
 
 export function setDashboardPanelContext(vizPanel: VizPanel, context: PanelContext) {
-  context.app = CoreApp.Dashboard;
+  let isEditable = false;
+
+  // @ts-ignore
+  let parent = vizPanel._parent;
+
+  while (parent != null) {
+    // @ts-ignore
+    if (parent._state?.isEditing) {
+      isEditable = true;
+      break;
+    }
+
+    // @ts-ignore
+    parent = parent._parent;
+  }
+
+  context.app = isEditable ? CoreApp.PanelEditor : CoreApp.Dashboard;
```

Comment:
Could we use URL for this?

Reply:
i considered it, but that tasted even worse :D

Reply:
Why not use same code as on line. 33 ?

Reply:
Why not use same code as on line. 33 ?

Reply:
cause i didnt scroll that far 🤣

Reply:
thanks, that works like my initial hack.

still there are some oddities:

1. i had to wrap + overwrite `dashboard.onEnterEditMode` to handle the case when you're not directly navigating to the edit page (such as clicking Edit panel)
2. there is no `dashboard.onExitEditMode` api, so i cannot revert the state, and when you exit out of panel edit, it retains the `CoreApp.PanelEditor` state.